src/test/java/com/puppycrawl/tools/checkstyle/filters/SuppressionXpathFilterTest.java

```diff
@@ -0,0 +1,203 @@
+            fail("Exception is expected");
+        }
+        catch (CheckstyleException ex) {
+            assertEquals("Invalid error message",
+                    "Unable to parse " + fileName + " - invalid files or checks format",
+                    ex.getMessage());
+        }
+    }
+
+    @Test
+    public void testExistingSuppressionFileWithTrueOptional() throws Exception {
+        final boolean optional = true;
+        final SuppressionXpathFilter filter =
+                createSuppressionXpathFilter(getPath("suppressions_none.xml"), optional);
+
+        final TreeWalkerAuditEvent ev = new TreeWalkerAuditEvent(null, "AnyJava.java", null, null);
+
+        assertTrue("Suppression file with true optional was not accepted",
+                filter.accept(ev));
+    }
+
+    @Test
+    public void testNonExistingSuppressionFileWithTrueOptional() throws Exception {
+        final String fileName = "src/test/resources/com/puppycrawl/tools/checkstyle/filters/"
+                + "non_existing_suppression_file.xml";
+        final boolean optional = true;
+        final SuppressionXpathFilter filter = createSuppressionXpathFilter(fileName, optional);
+
+        final TreeWalkerAuditEvent ev = new TreeWalkerAuditEvent(null, "AnyFile.java", null, null);
+
+        assertTrue("Should except event when suppression file does not exist",
+                filter.accept(ev));
+    }
+
+    @Test
+    public void testLocalFileExternalResourceContentDoesNotChange() throws Exception {
+        final DefaultConfiguration filterConfig = createModuleConfig(SuppressionXpathFilter.class);
+        filterConfig.addAttribute("file", getPath("suppressions_none.xml"));
+        final DefaultConfiguration treeWalkerConfig = createModuleConfig(TreeWalker.class);
+        treeWalkerConfig.addChild(filterConfig);
+
+        final DefaultConfiguration checkerConfig = new DefaultConfiguration("checkstyle_checks");
+        checkerConfig.addChild(treeWalkerConfig);
+        final String cacheFile = temporaryFolder.newFile().getPath();
+        checkerConfig.addAttribute("cacheFile", cacheFile);
+
+        final Checker checker = new Checker();
+        checker.setModuleClassLoader(Thread.currentThread().getContextClassLoader());
+        checker.addListener(getBriefUtLogger());
+        checker.configure(checkerConfig);
+
+        final String filePath = temporaryFolder.newFile("file.java").getPath();
+        final String[] expected = CommonUtils.EMPTY_STRING_ARRAY;
+
+        verify(checker, filePath, expected);
+        // One more time to use cache.
+        verify(checker, filePath, expected);
+    }
+
+    @Test
+    public void testReject() throws Exception {
+        final boolean optional = false;
+        final SuppressionXpathFilter filter = createSuppressionXpathFilter(
+                        getPath("suppressions_xpath_id_and_query.xml"), optional);
+        final File file = new File(getPath("InputSuppressionXpathFilter.java"));
```

Comment:
should be in the package suppressionxpathfilter

Reply:
done